### PR TITLE
Jg/streamtests

### PR DESCRIPTION
**Pull request by aneojgurhem**
- Improves the error message for tasks in error when the result is not available
- Stat to use channel in HtcMock instead of SubmitterClient in order to be able to use other clients.

fix https://github.com/aneoconsulting/ArmoniK/issues/756

### CODE REVIEW

File: Tests/Stream/Client/GrpcChannelExt.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+      var taskClient = new Tasks.TasksClient(channel);
+      var taskData = await taskClient.GetTaskAsync(new GetTaskRequest
+                                                   {
+                                                     TaskId = res.ResultTask.Single()
+                                                                 .TaskId,
+                                                   })
+                                     .ConfigureAwait(false);
+
+      if (taskData.Task.Output.Success)
```
Comment (jfonseca-aneo) [inline Tests/Stream/Client/GrpcChannelExt.cs:85]: I don't understand this line, why to throw on Success?

File: Tests/Stream/Client/StreamWrapperTests.cs [hunk 2]
```diff
@@ -69,13 +71,22 @@ public void SetUp()
...
+    channel_ = GrpcChannelFactory.CreateChannel(options);
+    client_  = new Submitter.SubmitterClient(channel_);
     Console.WriteLine("Client created");
   }
 
+  [TearDown]
+  public void TearDown()
+  {
+    partition_ = null;
+    client_    = null;
```
Comment (jfonseca-aneo) [inline Tests/Stream/Client/StreamWrapperTests.cs:83]: Do we know if `client_` is properly disposed ?

File: Tests/Stream/Client/GrpcChannelExt.cs (hunk 1 above)
Comment (aneojgurhem) [reply Tests/Stream/Client/GrpcChannelExt.cs:85]: We throw the original error (the one coming from the getresultasync)

File: Tests/Stream/Client/StreamWrapperTests.cs (hunk 2 above)
Comment (aneojgurhem) [reply Tests/Stream/Client/StreamWrapperTests.cs:83]: There are no dispose function in it

**Review by jfonseca-aneo — APPROVED** [review]

**Pull request merged**